addons/web/static/src/core/domain_selector/domain_selector.js

```diff
@@ -52,17 +50,22 @@ export class DomainSelector extends Component {
     }
 
     async onPropsUpdated(p) {
+        const classList = [
+            p.readonly ? "o_read_mode" : "o_edit_mode",
+            ...(p.className || "").split(" "),
+        ];
+        this.className = classList.filter((c) => c).join(" ");
```

Comment:
this.className = `${p.readonly ? "o_read_mode" : "o_edit_mode"} ${props.className}`.trim()

en getter 
+ default for className